lib/checkother.cpp

```diff
@@ -1793,7 +1793,10 @@ static bool isConstStatement(const Token *tok, bool cpp)
             const Token* lml = previousBeforeAstLeftmostLeaf(tok);
             if (lml)
                 lml = lml->next();
-            return lml && !isLikelyStream(cpp, lml) && isConstStatement(tok->astOperand2(), cpp);
+            const Token* stream = lml;
+            while (stream && Token::Match(stream->astParent(), ".|[|("))
```

Comment:
when parent is `[`  should we also check that `stream` is the lhs operator?

Reply:
In that case, `<<` is the astParent of `[`, which is checked by `isLikelyStream()`. Added a test case, tweaked condition.